### PR TITLE
bt-541: replace candidate screen with real data and connect filters

### LINKED ISSUE #541: Replace candidate screen with real data and connect filters
Image

### PR DESCRIPTION
**Pull request by mskmee**
Filters and cadidate taking data from backend.
It's because we didnt have bades with names from LMS.
image
image
image
image
image

### CODE REVIEW

File: mobile/src/bundles/employer/helpers/transform-candidate-filter-form-to-query/transform-candidate-filter-form-to-query.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+            result += `&englishLevel=${value}`;
+        }
+    }
+    if (employmentType?.length) {
+        for (const value of employmentType) {
+            result += `&employmentType=${value}`;
+        }
+    }
+
+    for (const [key, value] of Object.entries(multiSelectedData)) {
```
Comment (mskmee) [inline mobile/src/bundles/employer/helpers/transform-candidate-filter-form-to-query/transform-candidate-filter-form-to-query.ts:51]: thinking about another helper to simplify thoose for of, but dont knwo is it a good idea to create helper for helper?)

File: mobile/src/bundles/employer/components/candidate-card/candidate-card.tsx [hunk 2]
```diff
@@ -9,26 +9,44 @@ import {
...
+const getBadgeById = (data: BadgesItem[], id: string): BadgesItem => {
+    return data.find((value) => value.id === id) as BadgesItem;
+};
+
+const getHardSkillByValue = (
+    data: FormattedHardSkillsItem[],
+    id: string,
+): FormattedHardSkillsItem => {
+    return data.find(({ value }) => value === id) as FormattedHardSkillsItem;
+};
```
Comment (katerynakostikova) [inline mobile/src/bundles/employer/components/candidate-card/candidate-card.tsx]: can be moved to helpers

File: mobile/src/bundles/employer/employer-api.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        const response = await this.load(
+            this.getFullEndpoint(UserDetailsApiPath.ROOT, payload, {}),
+            {
+                method: 'GET',
+                contentType: ContentType.JSON,
+                hasAuth: true,
+            },
+        );
+        return await response.json<UserDetailsResponseDto[]>();
+    }
```
Comment (katerynakostikova) [inline mobile/src/bundles/employer/employer-api.ts]: I think it can be one request, filters will depend on query params

Comment (mskmee) [reply mobile/src/bundles/employer/employer-api.ts]: Than we will always run create query string function before make request, thougt that it's unnesseray. 
Combine to one request?

Comment (katerynakostikova) [reply mobile/src/bundles/employer/employer-api.ts]: It's ok

Comment (katerynakostikova) [reply mobile/src/bundles/employer/employer-api.ts]: Let's combine

Comment (katerynakostikova) [reply mobile/src/bundles/employer/employer-api.ts]: It's a common thing to have for example default query params, so it should be the same

**Review by katerynakostikova — APPROVED** [review]

**Comment by katerynakostikova** [comment]
Conflicts

**Review by anogire — APPROVED** [review]

**Pull request merged**